tests/test_freqtradebot.py

```diff
@@ -107,6 +107,7 @@ def test_order_dict_dry_run(default_conf, mocker, caplog) -> None:
     freqtrade = FreqtradeBot(conf)
     assert not freqtrade.strategy.order_types['stoploss_on_exchange']
     assert not log_has_re(".*stoploss_on_exchange .* dry-run", caplog)
+    caplog.clear()
```

Comment:
Can you explain what you're trying to reach with this (other than a lot of added lines)?

caplog is a fixture provided by pytest, which is at function scope - which means it's a "new" object for each new test, so clearing caplog at the end of a test is pointless.

While it's possible to clear caplog in the middle of a test (to have a clean sheet) which does make sense - at the end of a test i don't really see a point in it.